src/app/view/loadUtils.ts

```diff
@@ -31,15 +32,45 @@ export default class LoadUtils {
     static loadFromScratch(json: ScratchInterchangeV1): [Device, Array<RenderLayer>] {
         const newDevice: Device = LoadUtils.loadDeviceFromInterchangeV1(json);
         let newRenderLayers: Array<RenderLayer> = [];
-        for (let i = 0; i < json.renderLayers.length; i++) {
-            newRenderLayers.push(LoadUtils.loadRenderLayerFromInterchangeV1(json.renderLayers[i], newDevice));
+        if (json.renderLayers) {
+            for (let i = 0; i < json.renderLayers.length; i++) {
+                newRenderLayers.push(LoadUtils.loadRenderLayerFromInterchangeV1(json.renderLayers[i], newDevice));
+            }
+        } else if (Object.prototype.hasOwnProperty.call(json, "layers")) {
+            for (let i = 0; i < json.layers.length; i++) {
+                newRenderLayers.push(LoadUtils.generateRenderLayerFromLayerInterchangeV1(json.layers[i], newDevice));
+            }
+        } else {
+            newRenderLayers.push(new RenderLayer("flow", newDevice.layers[0], LogicalLayerType.FLOW));
+            newRenderLayers.push(new RenderLayer("control", newDevice.layers[1], LogicalLayerType.CONTROL));
+            newRenderLayers.push(new RenderLayer("integration", newDevice.layers[0], LogicalLayerType.INTEGRATION));
```

Comment:
Is render layer also gonna take a layer type ? s it even useful/necessary ?

Reply:
I set them to use `generateNewName` from device. However, this function does not take into account any objects which are loaded in. I am thinking of making a something in `loadFromScratch` in loadUtils which will determine the initial numbers for the addition of any object. I'm also unsure whether `generateNewName` should be removed from device or not, since it will also be handling RenderLayers, but the counts are specific to the device being worked on

Reply:
I've got a method set up that initializes the __nameMap in device when loading a JSON. It appears to be working; I've set it up for layers, connections, components, and renderLayers. Currently, it looks like features all get the same name. Let me know if you'd like me to do something regarding them.